### DISCUSSION TITLE
Radio button is not checked by default if values is numeric

### DISCUSSION THREAD

**Discussion by Phibee** [react-hook-form/react-hook-form#9549] [Q&A] [answered]
I have 2 sample created. one with the numeric values and one with string values. Noticed that one with numeric values unable to set checked or selected.  

sir @bluebill1049. Can you confirm if this is an issue?

https://codesandbox.io/s/brave-jennings-uqeqmj
https://codesandbox.io/s/xenodochial-visvesvaraya-4f6514

**Comment by Moshyfawn** [comment, marked as answer]
The HTML input value is always a `string`. So, passing `value={2}` is converted to `value="2"` _(you can see the input's value is a `string` by inspecting it with browser's DevTools inspector)_ As you're using uncontrolled inputs _(the [register](https://react-hook-form.com/api/useform/register) API)_, the source of truth for the input's value is.. well, the input and not a React state passed into the input, hence the "not selected" radio input as the default value is 2 of type `number` and the input's value is "2" of type `string` _(2 === "2" is `false`)_

**Reply by Moshyfawn** [reply]
So, not a bug

**Reply by bluebill1049** [reply]
thanks, @Moshyfawn I was just about to reply! ❤️

**Reply by Phibee** [reply]
Thanks @Moshyfawn , I get what you mean. Anyway yes you're right it is uncontrolled inputs. that's why I had to make use of controller like sample created below. :pray:

https://codesandbox.io/s/angry-feather-e48mnp?file=/src/App.js

**Reply by bluebill1049** [reply]
yes! that's right. I was going to follow it up, the beauty of a controlled component is input state is hosted via the local state, so it's much easier to customise the state shape and data type.
